src/dataflow/split_h5mu/script.py

```diff
@@ -0,0 +1,99 @@
+import mudata as mu
+import pandas as pd
+import re
+from pathlib import Path
+
+### VIASH START
+par = {
+  'input': 'harmony_knn/integrated.pynndescent_knn.output',
+  'modality': 'rna',
+  'obs_feature': 'dataset',
+  'output': 'reference_download/sample_split',
+  'drop_obs_nan': "true",
+  'output_compression': None,
+  'output_files': 'reference_download/sample_files.csv'
+}
+
+### VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
+
+
+def main():
+    logger.info(f"Reading {par['input']}")
+    input_file = Path(par["input"].strip())
+
+    mdata = mu.read_h5mu(input_file)
+    adata = mdata.mod[par["modality"]]
+
+    logger.info(f"Reading unique features from {par['obs_feature']}")
+    obs_features = adata.obs[par["obs_feature"]].unique().tolist()
+
+    # format obs_features into file compatible names
+    obs_features_s = [s.strip().replace(" ", "_") for s in obs_features]
+    obs_features_s = [re.sub(r'[-\s]', "_", s) for s in obs_features_s]
+    obs_features_s = [re.sub(r'[^A-Za-z0-9_]', "", s) for s in obs_features_s]
```

Comment:
I think this is shorter, because `/s` captures all whitespace characters (including spaces)
```suggestion
    obs_features_s = [re.sub(r'[-\s]', "_", s.strip()) for s in obs_features_s]
    obs_features_s = [re.sub(r'[^A-Za-z0-9_]', "", s) for s in obs_features_s]
```
Also, this currently only works when the `.obs` column only contains strings. But I think we should guard for:

1. NA (either `pd.NA` or `np.na`)
2. The `obs` not containing strings (but floats for example)